mlir/include/mlir/Dialect/MPI/IR/MPIOps.td

```diff
@@ -87,6 +109,37 @@ def MPI_SendOp : MPI_Op<"send", []> {
   let hasCanonicalizer = 1;
 }
 
+//===----------------------------------------------------------------------===//
+// ISendOp
+//===----------------------------------------------------------------------===//
+
+// TODO what about request handler?
```

Comment:
Conceptually, it should be a result of this op.

There are also some quirks of the design that this inherits from existing op that I'd model differently, but that's alright for now.

Reply:
I've added it as result. The question is whether we should add `MPI_Request_free` as an op.